src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/FeatureSourceIndexReader.java

```diff
@@ -62,25 +60,15 @@ public URIList getUriList(String profile, String timeField, String urlField, GoG
 
         query.setSortBy(new SortBy[]{sortByTimeAscending});
 
-        SimpleFeatureIterator iterator = null;
-        try {
-            iterator = getFeatures(typeName, query);
-        } catch (IOException e) {
-            throw new GoGoDuckException(e.getMessage());
-        }
-
-        try {
+        try (SimpleFeatureIterator iterator = getFeatures(typeName, query)) {
```

Comment:
can't use a for with SimplefeatureCollection - its a geotools collection not a real java collection.